### PR TITLE
USD -> SDF: Added lights attached to the world

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

## Summary

USD -> SDF: Added lights attached to the world

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: test/usd/upAxisZ.usda
Comment (koonpeng) [inline test/usd/upAxisZ.usda:27]: Is it intended that some of these attributes are undefined?

File: usd/src/usd_parser/USD2SDF.cc
Comment (koonpeng) [inline usd/src/usd_parser/USD2SDF.cc:85]: I'm confused about the role of this function.
* It looks like it is modifying the xml directly, shouldn't we use sdf apis instead?
* It looks like usd is already parsed into sdf by this point, why do we need this function?

File: usd/src/usd_parser/USDLight_TEST.cc
Comment (koonpeng) [inline usd/src/usd_parser/USDLight_TEST.cc:58]: Add some checks on the output?

File: usd/src/usd_parser/USDWorld.cc
Comment (koonpeng) [inline usd/src/usd_parser/USDWorld.cc:78]: It seems strange to me that `primPath` and `primType` are identical, maybe there is a mistake here?

File: usd/src/usd_parser/USDWorld.cc
Comment (koonpeng) [inline usd/src/usd_parser/USDWorld.cc:91]: The usd input may not come from isaac sim and so it may not be following isaac sim convention. We should probably note somewhere that the USD to SDF converter only works for isaac sim USDs.

**Review by koonpeng — CHANGES_REQUESTED** [review]

**Review by koonpeng — COMMENTED** [review]
code LGTM, but I got different results when converting sdf -> usd -> sdf

original:
```
    <light type="directional" name="sun">
      <cast_shadows>true</cast_shadows>
      <pose>0 0 10 0 0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.8 0.8 0.8 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>
```

usda
```
    def DistantLight "sun"
    {
        float inputs:intensity = 100
        float intensity = 100
        float3 xformOp:rotateXYZ = (0, -0, 0)
        double3 xformOp:translate = (0, 0, 10)
        uniform token[] xformOpOrder = ["xformOp:translate", "xformOp:rotateXYZ"]
    }
```

converted
```
    <light name='sun' type='directional'>
      <pose>0 0 10 0 -0 0</pose>
      <cast_shadows>true</cast_shadows>
      <intensity>0.01</intensity>
      <direction>0 0 -1</direction>
      <diffuse>1 1 1 1</diffuse>
      <specular>1 1 1 1</specular>
      <attenuation>
        <range>10</range>
        <linear>1</linear>
        <constant>1</constant>
        <quadratic>0</quadratic>
      </attenuation>
      <spot>
        <inner_angle>0</inner_angle>
        <outer_angle>0</outer_angle>
        <falloff>0</falloff>
      </spot>
    </light>
```

**Review by koonpeng — APPROVED** [review]
> There is some data that is not common between these two formats, I would say that some of the data might be "lost in translation"

I assume that these differences are due to the different formats and can't be addressed easily?

File: usd/src/usd_parser/USDWorld.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDWorld.cc:124]: Is this code mainly used for finding the parent prim of the light? I find it a little odd that we are manually parsing through the first 2 levels of the prim path to find the parent. I imagine this could be problematic if the light's parent isn't in the first two levels of the stage. Is there a better way to find the parent of a light that offers more flexibility/is less error prone?

**Review by adlarkin — COMMENTED** [review]
I have a few questions before merging. Otherwise, I think this looks good.

File: usd/src/usd_parser/USDWorld.cc
Comment (ahcorde) [reply usd/src/usd_parser/USDWorld.cc:124]: I didn't find a better way to do it, I'm open to suggestions

Comment (adlarkin) [reply usd/src/usd_parser/USDWorld.cc]: I can't think of a better way at the moment - since we are pressed for time, I added a TODO note in https://github.com/ignitionrobotics/sdformat/pull/875/commits/847271a1821043f64cefab13c0c7219a43d6cb6e and created an issue in #927.

**Pull request merged**
